db_abilities.py

```diff
@@ -70,13 +94,14 @@ def find_ability(self, ability_name):
 
 
 def get_or_make_ability(self, name):
-    cached = self.get_cached_ability(name)
-    if cached is not None:
-        return cached
     approximations, URL, path = self.find_ability(name)
     if len(approximations) != 1:
         raise AssertionError("Bad call to db_abilities.get_or_make_ability, incorrect number of approximations")
-    contents = webscraper.get_ability_contents(approximations[0], URL)
+    found_name = approximations[0]
+    cached = self.get_cached_ability(found_name)

```

Comment:
Check for approximations first to prevent bugs where we could cache the same ability twice if the ability name was not quite entered correctly but we still only found 1 approximation.
Checking for approximations should be relatively quick so this should be ok